src/builder/pdb.rs

```diff
@@ -0,0 +1,268 @@
+use crate::{
+    builder::ObjectMetaBuilder,
+    error::OperatorResult,
+    labels::{role_selector_labels, APP_MANAGED_BY_LABEL},
+    utils::format_full_controller_name,
+};
+use k8s_openapi::{
+    api::policy::v1::{PodDisruptionBudget, PodDisruptionBudgetSpec},
+    apimachinery::pkg::{
+        apis::meta::v1::{LabelSelector, ObjectMeta},
+        util::intstr::IntOrString,
+    },
+};
+use kube::{Resource, ResourceExt};
+
+#[derive(Debug, Default)]
+pub struct PdbBuilder<ObjectMeta, LabelSelector, Constraints> {
+    metadata: ObjectMeta,
+    selector: LabelSelector,
+    /// We intentionally only support fixed numbers, no percentage, see ADR 30 on Pod disruptions for details.
+    /// We use u16, as [`IntOrString`] takes an i32 and we don't want to allow negative numbers. u16 will always fit in i32.
+    max_unavailable: Option<u16>,
+    /// We intentionally only support fixed numbers, no percentage, see ADR 30 on Pod disruptions for details.
+    /// We use u16, as [`IntOrString`] takes an i32 and we don't want to allow negative numbers. u16 will always fit in i32.
+    min_available: Option<u16>,
+    /// Tracks wether either `max_unavailable` or `min_available` are set
+    _constraints: Constraints,
+}
+
+impl PdbBuilder<(), (), ()> {
+    pub fn new() -> Self {
+        PdbBuilder::default()
+    }
+
+    pub fn new_for_role<T: Resource<DynamicType = ()>>(
+        owner: &T,
+        app_name: &str,
+        role: &str,
+        operator_name: &str,
+        controller_name: &str,
+    ) -> OperatorResult<PdbBuilder<ObjectMeta, LabelSelector, ()>> {
+        let role_selector_labels = role_selector_labels(owner, app_name, role);
+        let metadata = ObjectMetaBuilder::new()
+            .namespace_opt(owner.namespace())
+            .name(format!("{}-{}", owner.name_any(), role))
+            .ownerreference_from_resource(owner, None, Some(true))?
+            .with_labels(role_selector_labels.clone())
+            .with_label(
+                APP_MANAGED_BY_LABEL.to_string(),
+                format_full_controller_name(operator_name, controller_name),
+            )
+            .build();
+        Ok(PdbBuilder {
```

Comment:
Let's space this out.

```suggestion
            .build();
            
        Ok(PdbBuilder {
```